libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -1883,17 +1861,47 @@ void GCS::update_receive(void)
 void GCS::send_mission_item_reached_message(uint16_t mission_index)
 {
     for (uint8_t i=0; i<num_gcs(); i++) {
-        if (chan(i).initialised) {
-            chan(i).mission_item_reached_index = mission_index;
-            chan(i).send_message(MSG_MISSION_ITEM_REACHED);
-        }
+        chan(i)->mission_item_reached_index = mission_index;
+        chan(i)->send_message(MSG_MISSION_ITEM_REACHED);
+    }
+}
+
+void GCS::setup_console()
+{
+    AP_HAL::UARTDriver *uart = AP::serialmanager().find_serial(AP_SerialManager::SerialProtocol_MAVLink, 0);
+    if (uart == nullptr) {
+        // this is probably not going to end well.
+        return;
     }
+    create_gcs_mavlink_backend(uart);
+}
+
+
+void GCS::create_gcs_mavlink_backend(AP_HAL::UARTDriver *uart)
+{
+    _chan[_num_gcs] = new_gcs_mavlink_backend(chan_parameters[_num_gcs], uart);
```

Comment:
We need to check that `_num_gcs` is less than the array size.

Reply:
Fixed